### DISCUSSION TITLE
Is there a way to associate an entity manager by attribute instead of directory?

### DISCUSSION THREAD

**Discussion by nesl247** [doctrine/DoctrineBundle#1579] [Q&A]
Hi!

If you have entities scattered across your `src` directory, and each belongs to it's own database, thus thus the need for multiple entity managers, how do you associate them to a particular entity manager? The only way I've seen is either via bundles or directories. However, directory doesn't work if you don't have a single directory for all of your entities, which is the case when using component not function based directory structures.

For example, the bundle expects `src/Entities` and for multiple EMs, `src/EM1/Entities` and `src/EM2/Entities`. However, we have `src/Component1/Entity1.php`, `src/Component2/Entity2.php`.

I was hoping for either a property on the `Entity` attribute called `entityManager` or `em`, or a separate attribute if necessary `EntityManager`, but I don't think this exists.

Thanks!

**Comment by stof** [comment]
there is nothing supported in doctrine/orm that allows such precise filtering

**Reply by nesl247** [reply]
@stof before I open a feature request then, do you feel that the request makes sense?

For everyone, I'm curious if anyone has solved this in some other way then.
